### PR TITLE
Migrate NuGetRestoreV1 task to Node10

### PR DESCRIPTION
**Pull request by mr-dokara**
**Task name**: NuGetRestoreV1

**Description**: Migrated NuGetRestoreV1 task to Node10.

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped
- [x] Checked that applied changes work as expected

Test results:
- Current canary tests for NuGet - [results](https://dev.azure.com/v-igolubev/Test/_build/results?buildId=46&view=results)
- Simple tests for NuGet restore - [results](https://dev.azure.com/v-igolubev/Test/_build/results?buildId=57&view=results)

### CODE REVIEW

**Review by DaniilShmelev — APPROVED** [review]

**Comment by anatolybolshakov** [comment]
@golubev-ivan could you please describe shortly which scenarios have you tested?

**Review by anatolybolshakov — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

**Comment by golubev-ivan** [comment]
> @golubev-ivan could you please describe shortly which scenarios have you tested?

I manually tested this task using a simple project without external packages.

**Comment by golubev-ivan** [comment]
cc @geekzter for visibility

**Comment by golubev-ivan** [comment]
Hi @aasim, @phil-hodgson, Could you please take a look at this PR?

**Comment by DmitriiBobreshev** [comment]
Test results are [here](https://github.com/microsoft/build-task-team/issues/3190)

**Comment by DmitriiBobreshev** [comment]
Hi @phil-hodgson, @aasim, @satba Could you please take a look at this PR?

**Review by tintse-thxsky-MSFT — CHANGES_REQUESTED** [review]
please look at the comment in package json

File: Tasks/NuGetRestoreV1/package.json [hunk 1]
```diff
@@ -17,10 +17,16 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "azure-pipelines-task-lib": "2.8.0",
-    "packaging-common": "file:../../_build/Tasks/Common/packaging-common-1.0.1.tgz",
```
[the remark this answers is not in the document]
Comment (mpodriezov) [reply Tasks/NuGetRestoreV1/package.json:21]: why do you think so @tintse-thxsky-MSFT ? We moved that as separate repos to not use file:/../../ aproach.

[the remark this answers is not in the document]
Comment (tintse-thxsky-MSFT) [reply Tasks/NuGetRestoreV1/package.json:21]: Never mind.  This is good.

**Review by tintse-thxsky-MSFT — APPROVED** [review]

**Review by tintse-thxsky-MSFT — APPROVED** [review]

**Pull request merged**